File: src/main/java/com/wesker926/leetcode/algorithms/p0237/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/11/2
// @description: 顶替
package s1

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteNode(node *ListNode) {
	node.Val = node.Next.Val
	node.Next = node.Next.Next
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0237/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0237.s1;

/**
 * @author wesker.gh
 * @date 2021/11/2
 * @description 顶替
 */
public class Solution {
    public void deleteNode(ListNode node) {
        node.val = node.next.val;
        node.next = node.next.next;
    }

    private static class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }
}
